src/routes/forgot-password/components/ForgotPassword.js

```diff
@@ -0,0 +1,106 @@
+const React = require('react');
+const T = require('prop-types');
+const StrangeForms = require('strange-forms');
+const IsEmail = require('utils/is-email');
+
+module.exports = class ForgotPassword extends StrangeForms(React.Component) {
+
+    static propTypes = {
+        requestReset: T.func.isRequired,
+        errorMessage: T.string
+    };
+
+    constructor(props) {
+
+        super(props);
+
+        this.state = {
+            email: '',
+            isBlurred: {
+                email: false
+            }
+        };
+
+        this.strangeForm({
+            fields: ['email'],
+            get: (someProps, field) => this.state[field],
+            act: this.formField.bind(this),
+            getFormValue: this.getFormValue.bind(this)
+        });
+    }
+
+    getFormValue(e) {
+
+        return e.target.value || '';
+    }
+
+    formField(field, value) {
+
+        this.setState({ [field]: value });
+    }
+
+    fieldBlurred = (ev) => {
+
+        const isBlurred = { ...this.state.isBlurred };
+        const field = ev.target.id;
+
+        isBlurred[field] = true;
+
+        this.setState({ isBlurred });
+    }
+
+    invalidEmail = () => {
+
+        if ( this.state.isBlurred.email) {
+
+            if ( IsEmail(this.state.email) ) {
+
+                return false;
+            }
+
+            return true;
+        }
+    }
```

Comment:
The fact that this can return nothing (and it's relied upon to not show the error message) isn't grabbing me well.  Maybe if this function is changed to something like `shouldShowErrorMessage`, and always returns a boolean value it would feel better?  I think it would also read better when used below.

Reply:
I think before I had it returning null (cause I agree, not returning something felt weird but didn't know if that was enough justification).